server/api.js

```diff
@@ -69,20 +74,183 @@ router.delete("/pages/:page_title/:record_id", async (req, res) => {
+	} finally {
+		await db.release();
+	}
+});
+
+router.post("/modules/imageAndTexts/:pageTitle", async (req, res) => {
+	const pageTitle = req.params.pageTitle;
+	const { text_header,text_body,image,button,hasbutton,imagetext_direction } = req.body;
 
+	if (
+		!text_header ||
+		!text_body ||
+		!image ||
+		!button ||
+		!hasbutton ||
+		!imagetext_direction
+	) {
+		res.status(500).json("Please fill all the fields");
+	}
+
+	try {
+		await db.query("begin");
+		// First inserting the textBanner table
+		const imageAndTexts = await db.query(
+			`INSERT INTO imageandtexts (text_header,text_body,image,button,hasbutton,imagetext_direction) 
+      VALUES ($1, $2, $3,$4,$5,$6)`,
+			[text_header, text_body, image, button, hasbutton, imagetext_direction]
+		);
+
+		//Second finding the pagge id
+		const findPageId = await db.query(
+			`select page_id from pages where page_title = $1`,
+			[pageTitle]
+		);
+		const page_id = findPageId.rows[0].page_id;
+		console.log(page_id);
+
+		// third finding the last record in the textBnner table
+		const lastRecord = await db.query(
+			`select record_id from imageAndTexts order by record_id desc limit 1`
+		);
+		const record_id = lastRecord.rows[0].record_id;
+
+		// fourth inserting the above datas into the modules table
+		const insertingModulesTable = await db.query(
+			`insert into modules(page_id,module_type,record_id) values($1,$2,$3)`,
+			[page_id, "imageAndTexts", record_id]
+		);
+		await db.query("commit");
+		return res.status(200).json({
+			message: `Data is stored successfuly into imageAndTexts table`,
+		});
+	} catch (error) {
+		console.error(error);
+		await db.query("rollback");
+		res.status(500).json({ error });
+	} finally {
+		await db.release();
+	}
+});
+
+router.post("/modules/heroBanner/:pageTitle", async (req, res) => {
+	const pageTitle = req.params.pageTitle;
+	const { heroImage, heroText } = req.body;
+
+	if (!heroImage ||!heroText ) {
+		res.status(500).json("Please fill all the fields");
+	}
+
+	try {
+		await db.query("begin");
+		// First inserting the textBanner table
+		const imageAndTexts = await db.query(
+			`INSERT INTO herobanner (hero_image,hero_text) 
+      VALUES ($1, $2)`,
+			[heroImage, heroText]
+		);
+
+		//Second finding the pagge id
+		const findPageId = await db.query(
+			`select page_id from pages where page_title = $1`,
+			[pageTitle]
+		);
+		const page_id = findPageId.rows[0].page_id;
+		console.log(page_id);
+
+		// third finding the last record in the textBnner table
+		const lastRecord = await db.query(
+			`select record_id from herobanner order by record_id desc limit 1`
+		);
+		const record_id = lastRecord.rows[0].record_id;
+
+		// fourth inserting the above datas into the modules table
+		const insertingModulesTable = await db.query(
+			`insert into modules(page_id,module_type,record_id) values($1,$2,$3)`,
+			[page_id, "heroBanner", record_id]
+		);
+		await db.query("commit");
+		return res.status(200).json({
```

Comment:
if you return here, it never gets to 'finally' block which means the db is never released

Reply:
Remove all the db transaction bloks